File: Shared/code/utility/BaseMessageDispatcher.cpp

#include "BaseMessageDispatcher.h"



using namespace google::protobuf;
using namespace BaseCmd;



BaseMessageDispatcher::BaseMessageDispatcher()
{


}

BaseMessageDispatcher::~BaseMessageDispatcher()
{
}

void BaseMessageDispatcher::Dispatch(std::shared_ptr<MessageRoute> spRoute,std::string typeName,std::shared_ptr<google::protobuf::Message> spMsg)
{
	if (m_ProcessorMap.find(typeName) != m_ProcessorMap.end())
	{
		m_ProcessorMap[typeName]->Process(spRoute,spMsg);
	}
}

File: AndroidProj/app/src/main/java/com/dd/androidprotobuf/BaseMessageDispatcher.java
package com.dd.androidprotobuf;

import com.google.protobuf.GeneratedMessageLite;

import java.util.Dictionary;
import java.util.Enumeration;
import java.util.HashMap;
import java.util.Map;

public class BaseMessageDispatcher {

    protected Map<String,BaseProcessor> m_ProcessorMap = new HashMap<String, BaseProcessor>();

    public void Dispatcher(MessageRoute route, String  typeName, GeneratedMessageLite msg)
    {
        if(m_ProcessorMap.containsKey(typeName))
        {
            m_ProcessorMap.get(typeName).Process(route,msg);
        }
    }
}
